### DISCUSSION TITLE
Accessing/Performing mysql query from codemagic.yaml

### DISCUSSION THREAD

**Discussion by danielccdk** [codemagic-ci-cd/codemagic-docs#2843] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

This is the following script in my codemagic.yaml file...
```
workflows:
  integration_test_workflow:
    scripts:
      - name: Query and delete appraisal
        script: |
          echo "Querying and deleting appraisal..."

          # # Install MySQL client
          sudo apt-get update && sudo apt-get install -y mysql-client

          # Query the last appraisal
          echo "Fetching the latest appraisal..."
          LATEST_APPRAISAL=$(mysql -h $DBHOST -u $DBUSER -p$DBPASS $DB -e "SELECT * FROM appraisals ORDER BY created_at DESC LIMIT 1;")

          # Echo the result to the console
          echo "Latest appraisal fetched from the database:"
          echo "$LATEST_APPRAISAL"
```
### Expected results

An output of the last item in the column.

### Actual results

Querying and deleting appraisal...
sudo: apt-get: command not found
Fetching the latest appraisal...
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_2_1_71w88t0h: line 10: mysql: command not found
Latest appraisal fetched from the database:

### Build id (optional)

673e4babfb5f444b87385065

**Comment by dam-ease** [comment]
Hello @danielccdk,
The Installation of MySQL client wasn't processed from the errors `sudo: apt-get: command not found` and `mysql: command not found`. To install the MySQL client using the YAML script, please try using the below script instead:
```
- name: Query and delete appraisal
        script: |
          echo "Querying and deleting appraisal..."

          # # Install MySQL client
          brew update
          brew install mysql-client
          mysql --version
          # Query the last appraisal
          echo "Fetching the latest appraisal..."
          LATEST_APPRAISAL=$(mysql -h $DBHOST -u $DBUSER -p$DBPASS $DB -e "SELECT * FROM appraisals ORDER BY created_at DESC LIMIT 1;")

          # Echo the result to the console
          echo "Latest appraisal fetched from the database:"
          echo "$LATEST_APPRAISAL"
```
Also, please confirm that the appraisals table exists and has records. If otherwise, the query will return no results and the $LATEST_APPRAISAL variable will be empty.

**Reply by danielccdk** [reply]
Hi my friend. Thank you so much for helping me...

Here is the result of using the script you had provided..

`
NOTE: ccache can prevent some software from compiling.
ALSO NOTE: The brew command, by design, will never use ccache.
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_3_1_0ztfp8wj: line 8: mysql: command not found
Fetching the latest appraisal...
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_3_1_0ztfp8wj: line 11: mysql: command not found
Latest appraisal fetched from the database:`

Thank you!

**Reply by dam-ease** [reply]
Hey @danielccdk,
Thanks for your response. A quick one here, please use `brew install mysql` instead. So the script will be as below:
```
- name: Query and delete appraisal
        script: |
          echo "Querying and deleting appraisal..."

          # # Install MySQL client
          brew update
          brew install mysql
          mysql --version
          # Query the last appraisal
          echo "Fetching the latest appraisal..."
          LATEST_APPRAISAL=$(mysql -h $DBHOST -u $DBUSER -p$DBPASS $DB -e "SELECT * FROM appraisals ORDER BY created_at DESC LIMIT 1;")

          # Echo the result to the console
          echo "Latest appraisal fetched from the database:"
          echo "$LATEST_APPRAISAL"
```

**Reply by dam-ease** [reply]
Hello @danielccdk,
Any update for me on this?If there is anything else I can assist you with, please don't hesitate to let me know.

**Reply by danielccdk** [reply]
We decided against it due to security issues. Sorry.
